shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -97,62 +99,68 @@ @interface FlutterEngine () <FlutterIndirectScribbleDelegate,
+@property(nonatomic, strong) FlutterMethodChannel* spellCheckChannel;
+@property(nonatomic, strong) FlutterBasicMessageChannel* lifecycleChannel;
+@property(nonatomic, strong) FlutterBasicMessageChannel* systemChannel;
+@property(nonatomic, strong) FlutterBasicMessageChannel* settingsChannel;
+@property(nonatomic, strong) FlutterBasicMessageChannel* keyEventChannel;
+@property(nonatomic, strong) FlutterMethodChannel* screenshotChannel;
 
 #pragma mark - Embedder API properties
 
 @property(nonatomic, assign) BOOL enableEmbedderAPI;
 // Function pointers for interacting with the embedder.h API.
 @property(nonatomic) FlutterEngineProcTable& embedderAPI;
+
 @end
 
 @implementation FlutterEngine {
-  fml::scoped_nsobject<FlutterDartProject> _dartProject;
   std::shared_ptr<flutter::ThreadHost> _threadHost;
   std::unique_ptr<flutter::Shell> _shell;
-  NSString* _labelPrefix;
-  std::unique_ptr<fml::WeakNSObjectFactory<FlutterEngine>> _weakFactory;
 
+  // TODO(cbracken): https://github.com/flutter/flutter/issues/155943
+  // Migrate to @property(nonatomic, weak).
   fml::WeakNSObject<FlutterViewController> _viewController;
-  fml::scoped_nsobject<FlutterDartVMServicePublisher> _publisher;
 
   std::shared_ptr<flutter::PlatformViewsController> _platformViewsController;
   flutter::IOSRenderingAPI _renderingApi;
   std::shared_ptr<flutter::ProfilerMetricsIOS> _profiler_metrics;
   std::shared_ptr<flutter::SamplingProfiler> _profiler;
 
-  // Channels
-  fml::scoped_nsobject<FlutterPlatformPlugin> _platformPlugin;
-  fml::scoped_nsobject<FlutterTextInputPlugin> _textInputPlugin;
-  fml::scoped_nsobject<FlutterUndoManagerPlugin> _undoManagerPlugin;
-  fml::scoped_nsobject<FlutterSpellCheckPlugin> _spellCheckPlugin;
-  fml::scoped_nsobject<FlutterRestorationPlugin> _restorationPlugin;
-  fml::scoped_nsobject<FlutterMethodChannel> _localizationChannel;
-  fml::scoped_nsobject<FlutterMethodChannel> _navigationChannel;
-  fml::scoped_nsobject<FlutterMethodChannel> _restorationChannel;
-  fml::scoped_nsobject<FlutterMethodChannel> _platformChannel;
-  fml::scoped_nsobject<FlutterMethodChannel> _platformViewsChannel;
-  fml::scoped_nsobject<FlutterMethodChannel> _textInputChannel;
-  fml::scoped_nsobject<FlutterMethodChannel> _undoManagerChannel;
-  fml::scoped_nsobject<FlutterMethodChannel> _scribbleChannel;
-  fml::scoped_nsobject<FlutterMethodChannel> _spellCheckChannel;
-  fml::scoped_nsobject<FlutterBasicMessageChannel> _lifecycleChannel;
-  fml::scoped_nsobject<FlutterBasicMessageChannel> _systemChannel;
-  fml::scoped_nsobject<FlutterBasicMessageChannel> _settingsChannel;
-  fml::scoped_nsobject<FlutterBasicMessageChannel> _keyEventChannel;
-  fml::scoped_nsobject<FlutterMethodChannel> _screenshotChannel;
-
-  int64_t _nextTextureId;
-
-  BOOL _allowHeadlessExecution;
-  BOOL _restorationEnabled;
   FlutterBinaryMessengerRelay* _binaryMessenger;
   FlutterTextureRegistryRelay* _textureRegistry;
```

Comment:
These two have public getters for `NSObject<FlutterBinaryMessenger>` and `NSObject<FlutterTextureRegistryRelay>`. 

I'd prefer to hold off on these to a future patch and they're alright as-is for now.